// wineitem/wineitem.component.ts
import { Component, OnInit } from '@angular/core';
import { Wine } from 'src/app/wine';

@Component({
  selector: 'app-wineitem',
  templateUrl: './wineitem.component.html',
  styleUrls: ['./wineitem.component.css']
})
export class WineitemComponent implements OnInit {
  //nombre, precio, imagen del vino, si está en venta, la cantidad comprada e incluso la lista de maridaje a la que puede acompañar este vino
  public wine: Wine | undefined;
  public wineClasses: any;


  constructor() { }

  ngOnInit(): void {
    this.wine.name = "Pepe";
    this.wine.price = 12.1;
    this.wine.isOnSale = true;
    this.wine.imageUrl = "https://www.mundovinum.com/8935-thickbox_default/vina-el-pison-2006.jpg";
    this.wine.quantityInCart = 10;
    this.wine.isOnSale = false;
  }

  addComprados(): void {
    if (this.wine.quantityInCart != undefined){
      this.wine.quantityInCart = this.wine.quantityInCart + 1;
    }
  }

  takeoutComprados(): void{
    if (this.wine.quantityInCart != undefined && this.wine.quantityInCart > 0 ){
      this.wine.quantityInCart = this.wine.quantityInCart - 1;
    }
  }
}
